### PR TITLE
Controller tests

### PR DESCRIPTION
**Pull request by henrikssn**
Adds controller tests.

We should start focusing on authentication tests to support #69.

Work in progress. Help is welcome.

### CODE REVIEW

**Comment by Zofz** [comment]
Will start here as well after CafeModule is live.

**Comment by Zofz** [comment]
The Rent tests have been changed in #52, does still need some controller-test.

File: app/models/candidate.rb [hunk 1]
```diff
@@ -25,10 +25,16 @@ def prepare(user)
...
+    user.present? && user.profile == self.profile
+  end
+
   def editable?
     return (self.election.view_status == 2) || (self.post.elected_by == "Studierådet")
   end
+
   def p_url
-    Rails.application.routes.url_helpers.election_candidate_url(self.id,host: PUBLIC_URL)
+    Rails.application.routes.url_helpers.election_candidate_url(self.id, host: PUBLIC_URL)
```
Comment (Zofz) [inline app/models/candidate.rb]: Will fix this and push with next commit (which will be tests for another controller).

**Comment by Zofz** [comment]
@henrikssn When improving tests for a model+controller it is hard to avoid changing a lot of connected code. This change might be very big.
Should we split it up into different pull request to this branch?

**Comment by henrikssn** [comment]
Fix the style then merge this and continue on same branch :)

**Comment by Zofz** [comment]
Merge it to master? Will do

**Comment by henrikssn** [comment]
Waaait! I will merge the migration fix first

File: config/environments/development.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # encoding: UTF-8
 Fsek::Application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
-	config.secret_key_base = 'd328a97ab11dafe94db121b97c79a3713cde6357d46c229c02bb082b9c6530f6a6c9a39b841b65d34fca35266d4684b1e4b06a0fdc018a55062f515e3ba7e6d2'
-  
-  
+  config.secret_key_base = 'd328a97ab11dafe94db121b97c79a3713cde6357d46c229c02bb082b9c6530f6a6c9a39b841b65d34fca35266d4684b1e4b06a0fdc018a55062f515e3ba7e6d2'
```
Comment (Zofz) [inline config/environments/development.rb]: Not gonna shorten this key, but we should move it somewhere else.

**Comment by Zofz** [comment]
Np, fixing hound atm.

File: spec/controllers/election/candidates_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                     candidate.lastname == change_attributes[:lastname]).to be_truthy
+        end
+
+        it "assigns the requested candidate as @candidate" do
+          patch :update, {id: candidate.to_param, candidate: change_attributes}
+          expect(assigns(:candidate)).to eq(candidate)
+        end
+
+        it "redirects to the candidate" do
+          patch :update, {id: candidate.to_param, candidate: change_attributes}
```
Comment (Zofz) [inline spec/controllers/election/candidates_controller_spec.rb]: This one is changed, but Hound won't update..

**Comment by Zofz** [comment]
This shit seriously never ends :O 
Is your fancy VIM-editor perfectly tuned for this already? ;)

**Comment by henrikssn** [comment]
This PR was deployed to Production. Reference: <a href='http://samson.fsektionen.se/projects/web/deploys/38' target='_blank'>d86e1fe</a>

**Pull request merged**